Pod/Classes/WPVideoPlayerView.m

```diff
@@ -41,18 +45,54 @@ - (void)commonInit {
     self.player = [[AVPlayer alloc] init];
     self.playerLayer = [AVPlayerLayer playerLayerWithPlayer: self.player];
     [self.layer addSublayer: self.playerLayer];
+    [self addSubview:self.controlToolbar];
+
+    __weak __typeof__(self) weakSelf = self;
+    self.timeObserver = [self.player addPeriodicTimeObserverForInterval:CMTimeMakeWithSeconds(1, NSEC_PER_SEC) queue:nil usingBlock:^(CMTime time) {
+        [weakSelf updateVideoDuration];
+    }];
 }
 
 - (void)dealloc {
-    [self.playerItem removeObserver:self forKeyPath: @"status"];
+    [_playerItem removeObserver:self forKeyPath: @"status"];
     [[NSNotificationCenter defaultCenter] removeObserver: self];
-    [self.player pause];
+    [_player removeTimeObserver:self.timeObserver];
+    [_player pause];
     _asset = nil;
+    _player = nil;
 }
 
 - (void)layoutSubviews {
     [super layoutSubviews];
     self.playerLayer.frame = self.bounds;
+    CGFloat position = toolbarHeight;
+    if (self.controlToolbarHidden) {
+        position = 0;
+    }
+    self.controlToolbar.frame = CGRectMake(0, self.frame.size.height - position, self.frame.size.width, toolbarHeight);
+}
+
+- (UIToolbar *)controlToolbar {
+    if (_controlToolbar) {
+        return _controlToolbar;
+    }
+    _controlToolbar = [[UIToolbar alloc] init];
+    _controlToolbar.hidden = YES;
+    [self updateControlToolbar];
+    return _controlToolbar;
+}
+
+- (UIBarButtonItem *)videoDurationButton {
+    if (_videoDurationButton == nil) {
+        _videoDurationButton = [[UIBarButtonItem alloc] initWithTitle:@"" style:UIBarButtonItemStylePlain target:nil action:nil];
```

Comment:
I saw an issue during testing where the button's text shifts slightly to the left or the right each time it updates its value. I just tried out switching things to use a right-aligned label with a fixed width, and it seems to behave much better. There's a diff here you can check out: https://gist.github.com/frosty/1d37e066224dc4b99551bcf4a1d29a49

The only thing I didn't add is updating the label's width on view rotation, which we should probably handle.

What do you think?